### PR TITLE
www-misc/buku: add 4.9

### PR DESCRIPTION
**Pull request by stkw0**
Closes: https://bugs.gentoo.org/929879

### CODE REVIEW

File: www-misc/buku/buku-4.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{9,10,11,12} )
```
Comment (ajakk) [inline www-misc/buku/buku-4.9.ebuild:7]: Please drop `python3_9`.

File: www-misc/buku/buku-4.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	newbashcomp auto-completion/bash/buku-completion.bash "${PN}"
+
+	doman buku.1
+}
+
+python_test() {
+	local skipped_tests=(
+		# Disable tests related to bukuserver
+		tests/test_setup.py::test_bukuserver_requirement
```
Comment (ajakk) [inline www-misc/buku/buku-4.9.ebuild:66]: I see there's a bunch more excluded tests from `test_views.py`. I haven't check, but surely that's only testing `bukuserver` bits, should we skip the whole file?

File: www-misc/buku/buku-4.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		>=dev-python/click-7.0[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		>=dev-python/py-1.5.0[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		>=dev-python/pyyaml-4.2[${PYTHON_USEDEP}]
+		>=dev-python/vcrpy-4.0.2[${PYTHON_USEDEP}]
+	)
+"
+
+PATCHES=( "${FILESDIR}/${P}-fix-py3.12.patch" )
```
Comment (ajakk) [inline www-misc/buku/buku-4.9.ebuild:40]: Hm, we need a patch for a linting? Why? We shouldn't be linting in tests in Portage, especially if it's broken.

File: www-misc/buku/buku-4.9.ebuild [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	insinto /usr/share/zsh/site-functions
+	doins auto-completion/zsh/_*
+
+	newbashcomp auto-completion/bash/buku-completion.bash "${PN}"
+
+	doman buku.1
+}
+
+python_test() {
+	local skipped_tests=(
```
Comment (ajakk) [inline www-misc/buku/buku-4.9.ebuild:64]: Probably tidier to switch to `distutils_enable_tests` + `EPYTEST_DESELECT`?

File: www-misc/buku/buku-4.9.ebuild (hunk 2 above)
Comment (stkw0) [reply www-misc/buku/buku-4.9.ebuild:66]: Thanks for the hint. Indeed, that file only contained bukuserver tests

**Comment by ajakk** [comment]
A nice bit tidier, thank you!

```
 ~/gentoo/gentoo/www-misc/buku $ wc -l buku-4.*
  91 buku-4.8.ebuild
  77 buku-4.9.ebuild
```

**Pull request closed without merging**